Ganache delays chain update on tx, which breaks unit tests (despite Ganache being up-to-date)

### Background / motivation

Issue [#339](https://github.com/oceanprotocol/barge/issues/339) first reported this. Its hypothesis was that a Ganache update would fix the issue. For the issue, Ganache was made up-to-date. [Here's the PR](https://github.com/oceanprotocol/barge/pull/342).

Yet we are still seeing this issue.
- I saw it on Dec 22, 2022, in ocean.py, and reported it as [ocean.py#1231](https://github.com/oceanprotocol/ocean.py/issues/1231).
- I saw it on Dec 25, 2022, in df-py, during work on [this PR](https://github.com/oceanprotocol/df-py/pull/408). The issue would go away if I created loops or other time-based delays; but of course these are not the fix, it just helps us understand the precise nature of the issue

### Ganache resources

- [trufflesuite.com/ganache](https://trufflesuite.com/ganache/)
  - [docs](https://trufflesuite.com/docs/ganache/)
    - feature: "Mine blocks instantly, on demand, or at an interval"
    - [CLI options](https://trufflesuite.com/docs/ganache/reference/cli-options/)
- [github.com/trufflesuite/ganache](https://github.com/trufflesuite/ganache)
  - [releases](https://github.com/trufflesuite/ganache/discussions/categories/releases). Ganache is currently at v7.7.2
  - [Ganache v7's Instamine Modes Explained](https://github.com/trufflesuite/ganache/discussions/2111)
  - [v7.3 bug fixes wrt instamine](https://github.com/trufflesuite/ganache/discussions/3209#user-content-7.3.0-fixes-2) "fix: save evm_mine blocks before returning [ganache#3016](https://github.com/trufflesuite/ganache/pull/3016)"

### Analysis / Potential directions

- In [PR for #339](https://github.com/oceanprotocol/barge/pull/342/files), it previously used the `istanbul` hardfork (old); and the PR _kept_ using that. The newest hardfork is called `merge`. Potential direction: use a newer hardfork 
- As of Ganache 7.3, there's an "instamine" feature. Are we using it? Are we susceptible to possible bugs? Investigate

- Experiment A:
  - in file "compose-files/ganache.yml", set hardfork = "grayGlacier"
  - in console 1: `./start_ocean.sh`
  - in console 2: `brownie test util/test_ganache/test_DFRewards.py::test_basic -s`
  - result: it was doing many many txs. After 100 s, still not done
- Analysis: this is weird. It should take 10 s max, with just a few txs


- Experiment B:
  - like experiment A, but with hardfork = "istanbul"
  - result: `INTERNALERROR>     raise ContractNotFound(f"No contract deployed at {address}")
INTERNALERROR> brownie.exceptions.ContractNotFound: No contract deployed at 0xf66870c3BE0533D1BF913Ac0C24D236dbA2a13C8`

- Experiment C:
  - deleted `~/.brownie/cache.db` and `deployments.db`
  - restarted ocean
  - then re-ran experiment B
  - result: like experiment B

- Experiment D:
  - deleted `~/df-py/build/`
  - then re-ran experiment B
  - result: it ran!
  - result: unit test passed. Total runtime 10 s
  - analysis: good, we're re-establishing baseline
  
- Experiment E:
  - deleted ~/df-py/build/
  - deleted ~/.brownie/cache.db and deployments.db
  - then re-ran experiment A (hardfork = "grayGlacier")
  - result:
```text
    def mint_fake_OCEAN(config: dict) -> None:
        """
        Does the following:
        1. Mints tokens
        2. Distributes tokens to TEST_PRIVATE_KEY1 and TEST_PRIVATE_KEY2
        """
        deployer_wallet = accounts.add(os.environ.get("FACTORY_DEPLOYER_PRIVATE_KEY"))
    
        OCEAN_token = Datatoken(config, address=get_ocean_token_address(config))
        amt_distribute = Web3.toWei("2000", "ether")
>       OCEAN_token.mint(
            deployer_wallet.address, Web3.toWei("20000", "ether"), {"from": deployer_wallet}
        )
E       ValueError: VM Exception while processing transaction: Transaction's maxFeePerGas (1) is less than the block's baseFeePerGas (18231706) (vm hf=grayGlacier -> block -> tx)

venv/lib/python3.10/site-packages/ocean_lib/ocean/mint_fake_ocean.py:26: ValueError
```


- Experiment F: try to fix grayGlacier fees
  - via editor, in df-py/util/test_ganache/conftest.py, add right after account creation: (inspiration from ocean.py, derived from eth-brownie docs)
    ```python
    from brownie.network import chain, priority_fee
    priority_fee(chain.priority_fee)
```
  - in console: `rm -rf ~/df-py/build/`
  - in console: `rm ~/.brownie/cache.db and deployments.db`
  - in file "compose-files/ganache.yml", set hardfork = "grayGlacier"
  - in console 1: `./start_ocean.sh`
  - in console 2: `brownie test util/test_ganache/test_DFRewards.py::test_basic -s`
  - result: it worked
  
  - via editor, in util/test_ganache/test_DFRewards.py::test_TOK, remove the time-based workaround
  - in console 2: `brownie test util/test_ganache/test_DFRewards.py::test_TOK -s`
  - result: it worked
  
  - in console 2: `brownie test util/test_ganache/test_DFRewards.py::test_OCEAN -s`
  - result: it worked

  - in console 2: run _all_ tests: `brownie test util/test_ganache/test_DFRewards.py`
  - result: failed test_TOK, test_multiple_TOK, test_bad_token, test_strategies
  - reasons for failure: the usual timing problem (where sleep() workaround fixes it) :(

  - summary: still have the timing-issues with "grayGlacier" fork

--- a/compose-files/ganache.yml
+++ b/compose-files/ganache.yml
@@ -7,4 +7,4 @@ services:
     networks:
       backend:
         ipv4_address: 172.15.0.3
-    entrypoint: ["node","/app/dist/node/cli.js","--database.dbPath", "./ganache_cache","--chain.chainId","0x2324","--chain.networkId","0x2324","-l","10000000000","-g","1","--chain.hardfork","istanbul","--wallet.mnemonic","${GANACHE_MNEMONIC}", "-e","1000000"]
+    entrypoint: ["node","/app/dist/node/cli.js","--database.dbPath", "./ganache_cache","--chain.chainId","0x2324","--chain.networkId","0x2324","-l","10000000000","-g","1","--chain.hardfork","istanbul","--miner.instamine", "eager","--wallet.mnemonic","${GANACHE_MNEMONIC}", "-e","1000000"]

--- a/compose-files/ganache.yml
+++ b/compose-files/ganache.yml
@@ -7,4 +7,4 @@ services:
     networks:
       backend:
         ipv4_address: 172.15.0.3
-    entrypoint: ["node","/app/dist/node/cli.js","--database.dbPath", "./ganache_cache","--chain.chainId","0x2324","--chain.networkId","0x2324","-l","10000000000","-g","1","--chain.hardfork","istanbul","--miner.instamine", "eager","--wallet.mnemonic","${GANACHE_MNEMONIC}", "-e","1000000"]
+    entrypoint: ["node","/app/dist/node/cli.js","--database.dbPath", "./ganache_cache","--chain.chainId","0x2324","--chain.networkId","0x2324","-l","10000000000","-g","1","--chain.hardfork","istanbul","--miner.blockTime","0","--miner.instamine", "eager","--wallet.mnemonic","${GANACHE_MNEMONIC}", "-e","1000000"]

